examples/format_seqFISH.py

```diff
@@ -0,0 +1,197 @@
+import functools
+import json
+import os
+from typing import Mapping, Tuple, Union
+
+import click
+import numpy as np
+import pandas as pd
+import skimage.io
+from slicedimage import ImageFormat
+
+from starfish.experiment.builder import FetchedTile, TileFetcher, write_experiment_json
+from starfish.types import Coordinates, Features, Indices, Number
+
+
+# We use this to cache images across tiles.  In the case of the osmFISH data set, volumes are saved
+# together in a single file.  To avoid reopening and decoding the TIFF file, we use a single-element
+# cache that maps between file_path and the npy file.
+@functools.lru_cache(maxsize=1)
+def cached_read_fn(file_path) -> np.ndarray:
+    return skimage.io.imread(file_path)
+
+
+class SeqFISHTile(FetchedTile):
+    def __init__(
+            self,
+            file_path: str,
+            coordinates: Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]],
+            z: int,
+            ch: int,
+    ):
+        self._file_path = file_path
+        self._z = z
+        self._ch = ch
+        self._coordinates = coordinates
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        """Gets image shape directly from the data"""
+        return self.tile_data().shape
```

Comment:
The downside of this approach is that you will decode the file twice.

Option 1: don't care, just decode twice.  easiest to read code at the cost of perf.
Option 2: cache the shape from the first tile, return it for all subsequent invocations of shape.  when `tile_data` is eventually called, verify that the shape matches the tile's shape.  this has significantly more complicated code.

Reply:
Given the `lru_cache` that's already in place around `imread`, the only thing that should be redone is the slicing of the loaded data by z and channel. That has a minimal performance loss. 

Am I interpreting the cache's functionality incorrectly?

Reply:
We read the shape on the first loop, and then we read the image data on a second loop.  This means we will read it twice.  As I said, this may not matter, but you should commit this code with that knowledge. :)